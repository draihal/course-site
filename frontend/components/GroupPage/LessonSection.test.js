import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import TestLessonSection from './LessonSection'

const lessons = [
  {
    "id": 1,
    "name": "lessontest",
    "slug": "lessontest",
    "number": 1,
    "description": "lessontest description",
    "poll_url": "http://test.com/",
    "datetime": "2020-02-10T20:00:05+03:00",
    "url_translation": "http://127.0.0.1:8000/",
    "homework_title": "lessontest homework_title",
    "homework_description": "lessontest homework_description",
    "homework_date": "2020-02-12",
    "module": 1,
    "group": 1
  }
];


describe('LessonSection', () => {
  it('render correctly LessonSection component', () => {
    const tree = renderer.create(<TestLessonSection lessons={lessons} />).toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('render LessonSection correctly with given props', () => {
    const { queryByText } = render(<TestLessonSection lessons={lessons} />);
    expect(queryByText(/lessontest description/i)).toBeTruthy();
  });

});