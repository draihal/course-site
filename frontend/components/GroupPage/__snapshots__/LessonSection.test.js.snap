// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LessonSection render correctly LessonSection component 1`] = `
<div
  className="col-md-12"
>
  <div>
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <div>
          <strong>
            Урок 
            1
            : 
            lessontest
          </strong>
        </div>
        <div>
          Описание: 
          lessontest description
        </div>
        <div>
          Дата проведения: 
          Mon Feb 10 2020 20:00:05 GMT+0300 (Moscow Standard Time)
        </div>
        <div>
          Ссылка на трансляцию: 
          <a
            href="http://127.0.0.1:8000/"
          >
            Перейти
          </a>
        </div>
      </div>
      <div
        className="col-md-6"
      >
        <div>
          <strong>
            Домашнее задание:
          </strong>
           
          lessontest homework_title
        </div>
        <div>
          Описание: 
          lessontest homework_description
        </div>
        <div>
          Сдать до: 
          2020-02-12
        </div>
        <div>
          Сдать: 
          <a
            href="/homework?lesson=1&title=lessontest homework_title"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Приступить
          </a>
        </div>
      </div>
    </div>
    <hr />
  </div>
</div>
`;
